static/js/notes.js

```diff
@@ -38,7 +100,30 @@ const $createNote = function(note) {
         .append($content);
 };
 
+// ADD NOTE
+const $createAddNoteModal = function() {
+    const $modal = $('<div>').addClass('modal');
+
+    const $buttons = NOTE_TYPES.map(noteType => {
+        return $createButtonWithText(noteType).addClass(classNamesByName.noteTypeButton);
+    });
+
+    const $textarea = $createTextarea('Note').addClass('note-content');
+
+    return $modal
+        .append($createGridContainer()
+            .append($createRowWithTwoElements($buttons[0], $buttons[1]))
+            .append($createRowWithTwoElements($buttons[2], $buttons[3])))
+        .append($textarea);
+};
+
+const activateModal = function($modal) {
+    mui.overlay('on', $modal.get(0));
```

Comment:
If i wanted to rename this code-ugliness what would you recommend i call it?  $obj.getRawDomObject()?

Reply:
I'm leaning heavily towards adding our own function to the jquery prototype to do this.

Reply:
Sorry for my vagueness. I dislike `$modal.get(0)`. I find it very unreadable, and counter intuitive. How do you know 0 is defined? What else is in the list? On top of that everytime i see it, my first thought is that the `$obj` is poorly named because i don't know what the `.get(0)` does. Then i remember that its this hack to get the `rawDomHtml` and i'm like oh when i have some extra time i should write a class to fix that.

Reply:
how does this look to you? https://github.com/atmiguel/cerealnotes/blob/10222d899fbd135cf5d0aed8210f4eea7c014626/static/js/base.js#L21 It means that in future code edits you can do `$modal.getDOM()`

Reply:
Why do you consider `.get()` a hack? `getDOM` is a bit confusing. Perhaps something like `getUnderlyingDOMElement`?

Reply:
`getUnderlyingDomElement` sounds great. Please note that i specifically didn't capilize `OM` in `DOM`. Its a convention at Microsoft which i now think is great. Simply put in camel case, everything is lowercase except for the first letter. This helps read the names clearly especially when there are multiple acronyms involved. Imaging you had a convention in your code to have ON mean Oriented North. Now this method is ambiguous `getJSONPut`. is it supposed to be `get JSON Put` or `get JS ON Put`? However, with the only one capital rule its clear. it'd be `getJsOnPut` or `getJsonPut` (granted you'd never want to have both of these methods in your code)